src/main/webapp/app/exercises/programming/manage/update/programming-exercise-plans-and-repositories-preview.component.ts

```diff
@@ -10,11 +15,69 @@ import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
 export class ProgrammingExercisePlansAndRepositoriesPreviewComponent {
     @Input() programmingExercise: ProgrammingExercise | null;
     @Input() isLocal: boolean;
+    @Input() programmingExerciseCreationConfig: ProgrammingExerciseCreationConfig;
 
-    getCourseShortName(): string | undefined {
+    constructor(private programmingExerciseService: ProgrammingExerciseService) {}
+
+    shortName?: string;
+
+    checkoutDirectories?: RepositoriesCheckoutDirectoriesDTO;
+    auxiliaryRepositoryCheckoutDirectories: string[] = [];
+
+    programmingExerciseServiceSubscription: Subscription;
+
+    private updateCheckoutDirectories() {
+        this.programmingExerciseServiceSubscription = this.programmingExerciseService
+            .getCheckoutDirectoriesForProgrammingLanguage(this.programmingExerciseCreationConfig.selectedProgrammingLanguage)
+            .subscribe((checkoutDirectories) => {
+                this.checkoutDirectories = checkoutDirectories;
+            });
+    }
+
+    private updateAuxiliaryRepositoryCheckoutDirectories() {
+        this.auxiliaryRepositoryCheckoutDirectories =
+            this.programmingExercise?.auxiliaryRepositories?.map((auxiliaryRepository) => addLeadingSlashIfNotPresent(auxiliaryRepository.checkoutDirectory)) ?? [];
+    }
+
+    private updateShortName() {
         if (!this.programmingExercise) {
-            return undefined;
+            return;
         }
-        return getCourseFromExercise(this.programmingExercise)?.shortName;
+        this.shortName = getCourseFromExercise(this.programmingExercise)?.shortName;
+    }
+
+    ngOnInit() {
```

Comment:
As far as I understand this check complains about the import `import { OnInit } from '@angular/core';` not being present.

IntelliJ will display it as unused method and complain about it, I do not see a benefit of this comment.